src/index.js

```diff
@@ -33,6 +33,22 @@ export function defaultMemoize(func, equalityCheck = defaultEqualityCheck) {
   }
 }
 
+export function resultCheckMemoize(func, equalityCheck = defaultEqualityCheck, resultCheck = defaultEqualityCheck) {
+  let lastArgs = null
+  let lastResult = null
+  // we reference arguments instead of spreading them for performance reasons
+  return function () {
+    if (areArgumentsShallowlyEqual(equalityCheck, lastArgs, arguments)) {
+      // apply arguments instead of spreading for performance.
+      return lastResult
+    }
+
+    lastArgs = arguments
+    const result = func.apply(null, arguments)
```

Comment:
Arguably, resultCheck shouldn't get called the first time?

Reply:
Agreed!